src/shared/utils/rxjs.ts

```diff
@@ -0,0 +1,23 @@
+import { Observable, Subject, finalize, mergeWith, takeUntil } from "rxjs";
+
+/**
+ * Merges two observables until the first one completes (or errors).
+ * The difference between this and `merge` is that this will complete when the first observable completes,
+ * while `merge` would only complete when _all_ observables complete.
+ */
+export function mergeUntilFirstComplete<FirstObsType, SecondObsType>(
+  observable1: Observable<FirstObsType>,
+  observable2: Observable<SecondObsType>,
+): Observable<FirstObsType | SecondObsType> {
+  const completionSubject = new Subject<void>();
+  const observable1WithCompletion = observable1.pipe(
+    takeUntil(completionSubject),
+    finalize(() => completionSubject.next()),
```

Comment:
Style: `completionSubject.complete()` would have the same effect, but would be easiert to understand at first glance, I had to spend a minute or two to understand what's going on here.

Reply:
`completionSubject.complete` doesn't trigger `takeUntil` to stop, only `next`.